sstables/sstable_set.hh

```diff
@@ -112,7 +112,7 @@ public:
         schema_ptr,
         reader_permit,
         utils::estimated_histogram&,
-        const dht::partition_range&, // must be singular
+        const dht::ring_position&,
```

Comment:
I'd have to extract the key where I call this function, and inside this function I would have to construct a `ring_position` again (because `read_row_flat` and `select` expect a `ring_position`), which requires copying the key. I'd like to avoid the copying if possible (may be expensive for large keys).

Changing from `partition_range` to `ring_position` is still an improvement and requires minimal effort. I'm guessing that changing into `partition_key` without having to copy would require a deeper refactor.

Reply:
I see. Those functions should accept a `ring_position_view` (which we don't have), but that of course is outside the scope of this patch.
Indeed this is an improvement over the previous situation.